lms/djangoapps/certificates/views/webview.py

```diff
@@ -588,46 +546,131 @@ def render_html_view(request, user_id, course_id):
-    context.update(course.cert_html_view_overrides)
+        # Append/Override the existing view context values with certificate specific values
+        _update_certificate_context(context, course, user_certificate, platform_name)
 
-    # Track certificate view events
-    _track_certificate_events(request, context, course, user, user_certificate)
+        # Append badge info
+        _update_badge_context(context, course, user)
 
-    # FINALLY, render appropriate certificate
-    return _render_certificate_template(request, context, course, user_certificate)
+        # Append site configuration overrides
+        _update_configuration_context(context, configuration)
+
+        # Add certificate header/footer data to current context
+        context.update(get_certificate_header_context(is_secure=request.is_secure()))
+        context.update(get_certificate_footer_context())
+
+        # Append/Override the existing view context values with any course-specific static values from Advanced Settings
+        context.update(course.cert_html_view_overrides)
+
+        # Track certificate view events
+        _track_certificate_events(request, context, course, user, user_certificate)
+
+        # Render the certificate
+        return _render_valid_certificate(request, context, custom_template)
+
+
+def _get_catalog_data_for_course(course_key):
+    """
+    Retrieve data from the Discovery service necessary for rendering a certificate for a specific course.
+    """
+    course_certificate_settings = CertificateGenerationCourseSetting.get(course_key)
+    if not course_certificate_settings:
+        return {}
+
+    catalog_data = {}
+    course_run_fields = []
+    if course_certificate_settings.language_specific_templates_enabled:
+        course_run_fields.append('content_language')
+    if course_certificate_settings.include_hours_of_effort:
+        course_run_fields.extend(['weeks_to_complete', 'max_effort'])
+
+    if course_run_fields:
+        course_run_data = get_course_run_details(course_key, course_run_fields)
+        if course_run_data.get('weeks_to_complete') and course_run_data.get('max_effort'):
+            try:
+                weeks_to_complete = int(course_run_data['weeks_to_complete'])
+                max_effort = int(course_run_data['max_effort'])
+                catalog_data['hours_of_effort'] = weeks_to_complete * max_effort
+            except ValueError:
+                log.exception('Error occurred while parsing course run details')
+        catalog_data['content_language'] = course_run_data.get('content_language')
+
+    return catalog_data
+
+
+def _get_closest_released_language(certificate_language):
+    """
+    Look for a released language that is compatible with the two-digit certificate language and return it, otherwise
+    return the original certificate language code.
+    """
+    for language in released_languages():
+        if language.code[:2] == certificate_language:
```

Comment:
Is this something we'd care about if we had en-US and en-UK for example? Or would just a default "en" good enough even if some spellings are off?

Reply:
Shouldn't we look for the exact match first and then fall back to the 2 letter code?